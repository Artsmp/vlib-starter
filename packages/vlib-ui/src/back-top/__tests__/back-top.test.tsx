import { nextTick } from 'vue'
import { mount } from '@vue/test-utils'
import { describe, test, expect } from 'vitest'
import { BackTop } from '../index'

describe('BackTop.vue', async () => {
  const wrapper = mount(
    () => {
      return (
        <div class={'target'} style={{ height: 100, overflow: 'auto' }}>
          <div style={{ height: 10000 }}>
            <BackTop
              target=".target"
              visibilityHeight={300}
              bottom={100}
              right={100}
            />
          </div>
        </div>
      )
    },
    { attachTo: document.body }
  )

  await nextTick()
  test('初始时隐藏', () => {
    // 初始时隐藏
    expect(wrapper.find('.vlib-back-top').exists()).toBe(false)
  })
  test('滚动到2000后元素显示', async () => {
    wrapper.element.scrollTop = 2000
    await wrapper.trigger('scroll')
    expect(wrapper.find('.vlib-back-top').exists()).toBe(true)
  })
  test('验证props是否渲染正确', async () => {
    expect(wrapper.find('.vlib-back-top').attributes('style')).toBe(
      'right: 100px; bottom: 100px;'
    )
  })
  test('点击返回触发 click 事件', async () => {
    await wrapper.trigger('click')
    expect(wrapper.emitted('click')).toBeDefined()
  })
  test('输出快照到单独的文件', async () => {
    // 输出快照到单独的文件
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('target不存在', () => {
    // const fn = () => {
    //   mount(
    //     () => {
    //       return (
    //         <div class={'target'} style={{ height: 100, overflow: 'auto' }}>
    //           <div style={{ height: 10000 }}>
    //             <BackTop
    //               target=".notExist"
    //               visibilityHeight={300}
    //               bottom={100}
    //               right={100}
    //             />
    //           </div>
    //         </div>
    //       )
    //     },
    //     { attachTo: document.body }
    //   )
    // }
    // expect(fn()).toThrowErrorMatchingInlineSnapshot()
  })
})
