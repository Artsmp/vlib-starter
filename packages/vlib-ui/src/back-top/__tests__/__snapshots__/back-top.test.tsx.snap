// Vitest Snapshot v1

exports[`BackTop.vue > render 1`] = `
"<div class=\\"target\\" style=\\"height: 100; overflow: auto;\\">
  <div style=\\"height: 10000;\\">
    <transition-stub name=\\"van-fade\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
      <div class=\\"vlib-back-top\\" style=\\"right: 100px; bottom: 100px;\\"><button type=\\"button\\" class=\\"van-button van-button--primary van-button--normal van-button--round\\" vlib-back-top=\\"\\">
          <div class=\\"van-button__content\\"><i class=\\"van-badge__wrapper van-icon van-icon-back-top van-button__icon\\">
              <!---->
              <!---->
              <!---->
            </i>
            <!---->
            <!---->
          </div>
        </button></div>
    </transition-stub>
  </div>
</div>"
`;

exports[`BackTop.vue > 输出快照到单独的文件 1`] = `
"<div class=\\"target\\" style=\\"height: 100; overflow: auto;\\">
  <div style=\\"height: 10000;\\">
    <transition-stub name=\\"van-fade\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
      <div class=\\"vlib-back-top\\" style=\\"right: 100px; bottom: 100px;\\"><button type=\\"button\\" class=\\"van-button van-button--primary van-button--normal van-button--round\\" vlib-back-top=\\"\\">
          <div class=\\"van-button__content\\"><i class=\\"van-badge__wrapper van-icon van-icon-back-top van-button__icon\\">
              <!---->
              <!---->
              <!---->
            </i>
            <!---->
            <!---->
          </div>
        </button></div>
    </transition-stub>
  </div>
</div>"
`;
